### ISSUE TITLE
`reportInconsistentOverload` false negative when implementation has decorator that uses a paramspec

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#1665]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportUnusedParameter=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCAhgDYlEBGJApgDRRgBu1IJYRAJgLABQvvHasCiCAxmADaAKikAFegBUAugApgKAFyFS5KtQnyoygJRQAtAD5tZSjQOKlWgHQv%2BfHgAEmLNpwFCoYDAwFSIAZw5gLVQYU0tMFBhnV3deL2ZWdm4eQWEgkPDIrTCYEDirEpBkpzc0sTB-POCVY2qgA)

```python
from typing import Callable, overload


def deco[**P, T](fn: Callable[P, T]) -> Callable[P, T]: ...


@overload
def foo(asdf: int) -> int: ...


@overload
def foo(asdf: str) -> str: ...


@deco
def foo(): ... # no error
```
